datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermUpdateInputSnapshotMapper.java

```diff
@@ -0,0 +1,85 @@
+
+    public static final GlossaryTermUpdateInputSnapshotMapper INSTANCE = new GlossaryTermUpdateInputSnapshotMapper();
+
+    public static GlossaryTermSnapshot map(
+            @Nonnull final GlossaryTermUpdateInput glossaryTermUpdateInput,
+            @Nonnull final Urn actor) {
+        return INSTANCE.apply(glossaryTermUpdateInput, actor);
+    }
+
+    @Override
+    public GlossaryTermSnapshot apply(
+            @Nonnull final GlossaryTermUpdateInput glossaryTermUpdateInput,
+            @Nonnull final Urn actor) {
+        final GlossaryTermSnapshot result = new GlossaryTermSnapshot();
+        final AuditStamp auditStamp = new AuditStamp();
+        auditStamp.setActor(actor, SetMode.IGNORE_NULL);
+        auditStamp.setTime(System.currentTimeMillis());
+        try {
+            GlossaryTermUrn glossaryTermUrn = GlossaryTermUrn.createFromString(glossaryTermUpdateInput.getUrn());
+            result.setUrn(glossaryTermUrn);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException(
+                    String.format("Failed to validate provided urn with value %s", glossaryTermUpdateInput.getUrn()));
+        }
+        final GlossaryTermAspectArray aspects = new GlossaryTermAspectArray();
+
+        if (glossaryTermUpdateInput.getEditableSchemaMetadata() != null) {
+            final EditableSchemaMetadata editableSchemaMetadata = new EditableSchemaMetadata();
+            editableSchemaMetadata.setEditableSchemaFieldInfo(
+                    new EditableSchemaFieldInfoArray(
+                            glossaryTermUpdateInput.getEditableSchemaMetadata().getEditableSchemaFieldInfo().stream().map(
+                                    element -> mapSchemaFieldInfo(element)
+                            ).collect(Collectors.toList())));
+            editableSchemaMetadata.setLastModified(auditStamp);
+            editableSchemaMetadata.setCreated(auditStamp);
+            aspects.add(GlossaryTermAspect.create(editableSchemaMetadata));
+        }
+        result.setAspects(aspects);
+
+        return result;
+    }
+
+    private EditableSchemaFieldInfo mapSchemaFieldInfo(
+            final com.linkedin.datahub.graphql.generated.EditableSchemaFieldInfoUpdate schemaFieldInfo
+    ) {
+        final EditableSchemaFieldInfo output = new EditableSchemaFieldInfo();
+
+        if (schemaFieldInfo.getDescription() != null) {
+            output.setDescription(schemaFieldInfo.getDescription());
+        }
+        output.setFieldPath(schemaFieldInfo.getFieldPath());
+
+        if (schemaFieldInfo.getGlobalTags() != null) {
+            final GlobalTags globalTags = new GlobalTags();
+            globalTags.setTags(new TagAssociationArray(schemaFieldInfo.getGlobalTags().getTags().stream().map(
+                    element -> TagAssociationUpdateMapper.map(element)).collect(Collectors.toList())));
+            output.setGlobalTags(globalTags);
+        }
+        return output;
+    }
```

Comment:
shouldn't this be a shared util? why do we need to rewrite this

Reply:
removed